From 04d5964bb4caef4a9d131828ebaf6967e1e29b73 Mon Sep 17 00:00:00 2001
From: rikardo <rikard.olander@hostmobility.com>
Date: Thu, 2 Mar 2023 16:14:20 +0000
Subject: [PATCH 1/1] imx8mp-var-dart: reduce I2C1 clock frequency per errata

Lower I2C1 clock-frequency from 400 kHz to 384 kHz to comply with
i.MX8MP errata (IMX8MP_1P33A rev2.0) which specifies maximum I2C
clock frequency of 384 kHz.

Remove GPIO recovery configuration (scl-gpios, sda-gpios, and gpio
pinctrl) as it's not needed and causes boot warnings.

Signed-off-by: Rikard Olander <rikard.olander@hostmobility.com>
Signed-off-by: Andreas Ternstedt <a.ternstedt@setek.com>

---
 arch/arm64/boot/dts/freescale/imx8mp-var-dart.dtsi | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-var-dart.dtsi b/arch/arm64/boot/dts/freescale/imx8mp-var-dart.dtsi
index c64cbe80e620..31ecea223945 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-var-dart.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mp-var-dart.dtsi
@@ -147,12 +147,9 @@ &gpu_reserved {
 };
 
 &i2c1 {
-	clock-frequency = <400000>;
-	pinctrl-names = "default", "gpio";
+	clock-frequency = <384000>;
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c1>;
-	pinctrl-1 = <&pinctrl_i2c1_gpio>;
-	scl-gpios = <&gpio5 14 GPIO_ACTIVE_HIGH>;
-	sda-gpios = <&gpio5 15 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
 	pca9450@25 {
-- 
2.34.1

